### PR TITLE
Python: test the MaD path for constructor calls

### PR DESCRIPTION
**Pull request by yoff**
We currently can identify calls to class constructors (to the `__init__` method) via the `Call` token.
It is not clear that this is how we want that to look, but we need a way and currently have one, so these tests document that.

### CODE REVIEW

File: python/ql/test/experimental/dataflow/model-summaries/NormalDataflowTest.ext.yml [hunk 1]
```diff
@@ -18,4 +18,6 @@ extensions:
       - ["foo", "Member[MS_spread]", "Argument[0]", "ReturnValue.TupleElement[0]", "value"]
       - ["foo", "Member[MS_spread]", "Argument[1]", "ReturnValue.TupleElement[1]", "value"]
       - ["foo", "Member[MS_spread_all]", "Argument[0]", "ReturnValue.TupleElement[0,1]", "value"]
+      - ["foo", "Member[MS_Class].Call", "Argument[0, x:]", "ReturnValue", "value"]
```
Comment (RasmusWL) [inline python/ql/test/experimental/dataflow/model-summaries/NormalDataflowTest.ext.yml:21]: I suggest you rewrite this to store the Argument 0/x: into an attribute of the return value, that's more representative of what would usually happen in a model (I assume).

**Review by RasmusWL — CHANGES_REQUESTED** [review]

File: python/ql/test/experimental/dataflow/model-summaries/NormalDataflowTest.ext.yml (hunk 1 above)
Comment (yoff) [reply python/ql/test/experimental/dataflow/model-summaries/NormalDataflowTest.ext.yml]: Done. Also updated to latest main. Sorry for the force push; fortunately the changes are quite contained.

**Review by RasmusWL — COMMENTED** [review]
Looking over this PR again, I realize we should also do a non-trivial subclass. So something like

```
class D(MS_Class_transitive):
    def __init__(x, y):
        # special handling of y
        super().__init__(x)

class E(MS_Class_transitive):
    def __init__(y, x):
        # special handling of y
        super().__init__(x)
```

we could handle both properly, but won't handle E properly right now :( -- and I think that's worth including in the test

**Comment by yoff** [comment]
> we could handle both properly, but won't handle E properly right now :( -- and I think that's worth including in the test

Ah, I did not realise that. Test added 👍

**Review by RasmusWL — APPROVED** [review]

**Comment by yoff** [comment]
Force pushed since conflicts were resolved cleanly by `pull --rebase`.

**Review by RasmusWL — APPROVED** [review]

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**